### ISSUE TITLE
Regression : Opengl 2.0 crashes OsmAnd and is very slow

### ISSUE THREAD

**Issue by Rpnpif** [osmandapp/OsmAnd#20190]
### Description

The 4.7.10 and 4.7.17 version does crash OsmAnd when OpenGL 2 is enabled. OpenGL 1 was very slow.

### Steps to reproduce

Enable OpenGL 2.
Open a map.
Browse some view of the screen.
Try OpenGL 1.

### Actual result

OpenGL 2 : After on some seconds, the rendering is too slow and crash. This is unusable.
OpenGL 1 : All work but less but still too slowly with some freeze during one to three seconds. This is usable but the CPU is too hot.

4.6 version works fine but it is still too slow without crashing.

### Expected result

As with the first 4.x releases, OpenGL 2 worked fine with acceptable speed of the rendering. So it was a regression.

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version: 4.7.10 471003
Android/iOS version: 8.0.0
Device model: HUAWEI HWPRA-H
Crash-Logs: 
```

**Comment by DmitryAlexei** [comment]
Have you tried to disable and delete live updates > clear app cash > enable and redownload map updates?

**Comment by Rpnpif** [comment]
@DmitryAlexei You're right. Disabling live makes it work fine. And the speed of the rendering is acceptable with OpenGL.

As soon as I reactivate live even after deleting old files, the malfunctions and crashes return. It's annoying because I need it to evaluate recent modifications for OSMAnd in OpenStreetMap.

So 4.7+ is incompatible with Live for me.

Thank you for the workaround which permits me to use OsmAnd, but the issue is still there.

**Comment by yuriiurshuliak** [comment]
This bug was fixed  last week. Please update to the latest version or join the beta version at https://osmand.net/docs/versions/nightly_versions#google-play-beta-builds
Additionally, ensure that live updates are disabled and deleted, clear the cache, then enable and redownload the updates.

**Comment by yuriiurshuliak** [comment]
Duplicate of https://github.com/osmandapp/OsmAnd/issues/19890

**Comment by yuriiurshuliak** [comment]
The bug has been reproduced. For reproduction, I downloaded the map of Moscow and tested its functionality from the start without live updates. Then, I enabled live updates for this map and tested it again. During the testing, I observed that the map started to load more slowly when changing the zoom level and experienced lag when dragging the map.

After further use, the application crashed. Here are the logs: 
[exception (16).zip](https://github.com/user-attachments/files/16054511/exception.16.zip)

OsmAnd~ 4.8.0#2643m, released: 2024-07-01

https://github.com/osmandapp/OsmAnd/assets/127092082/39eb27ec-afeb-46ce-9e2c-1f877aa634bb

**Comment by tomchiverton** [comment]
Is the intent to ship a fix that fides this automatically? Most people aren't going to come here or stumble on the workaround of the other render mode, yet alone removing all their local maps while just the right settings are selected.

**Comment by vshcherb** [comment]
the problem was related gto vandalism it's hard to revertn change in the middle especially in the middle of the month. we're striving to prepare new monthly maps soon to start live updates from scratch

**Comment by rei-zero246** [comment]
Is this crash related?

```
type: crash
osVersion: google/husky/husky:14/AP2A.240705.005/2024070201:user/release-keys
uid: 10179 (u:r:untrusted_app_32:s0:c179,c256,c512,c768)
cmdline: net.osmand.plus
processUptime: 4s

signal: 11 (SIGSEGV), code 9 (SEGV_MTESERR), faultAddr 900bed9c66f3380
threadName: Worker (pooled)
MTE: enabled

backtrace:
    /data/app/~~SmqkyQYXG9oM3RhFS3hALA==/net.osmand.plus-ai1WcvD9bkdk6RC65Zh5mg==/lib/arm64/libQt5Core.so (pc 54f49c)
    <anonymous:c1ab58e35000> (pc fb58)
```

**Comment by Rpnpif** [comment]
@vshcherb. You are right. After an updating of the maps, Live works fine now.
But this bug will recur in the event of new vandalism, so a fix is needed.

**Comment by vshcherb** [comment]
We have a fix on server side for next time

**Comment by dklymenk** [comment]
It appears that the current solutions are either use Version 1 rendering engine or stop using live updates. Is there another GitHub issue that tracks the live update incompatibility with Version 2 (OpenGL) rendering engine?

I'm on OsmAnd~ 4.8.5 downloaded from F-Droid, freshly installed after an uninstall, and I am still experiencing the issue with live updates not showing up when using OpenGL.

For example, there is this [node](https://www.openstreetmap.org/node/12115777281) I added [17 days ago](https://www.openstreetmap.org/node/12115777281/history/1). It's there on the web map, I can see it with the Version 1 renderer, it's there on the CyclOSM overlay, but, no matter what I do, the node doesn't show up when using Version 2 renderer.

Attaching screenshots and some extra information potentially useful for troubleshooting.

<details>
<summary>Expand for screenshots</summary>

Map of the region (Canada, Ontario, Kingston) dated 2024-07-31:
![image](https://github.com/user-attachments/assets/d7c62f1b-0eac-4881-8d51-de4b14f913d4)

Live updates enabled
![image](https://github.com/user-attachments/assets/aa92f278-e52c-44a8-9dd3-be909bdfad50)

Map updates for mappers enabled
![image](https://github.com/user-attachments/assets/e5eac777-5745-4441-8485-aa81c931ba90)

Version 1 rendering engine
![image](https://github.com/user-attachments/assets/db2ea1c0-7d11-439d-b9cb-9bafb9606171)

Version 2 rendering engine
![image](https://github.com/user-attachments/assets/9ea35c42-2494-4dc7-948b-68f524e3dee6)


</details>

**Comment by DmitryAlexei** [comment]
@dklymenk Please update the app to the latest version (4.8.6). For correct work of live updates use the latest standard maps available.

**Comment by dklymenk** [comment]
> @dklymenk Please update the app to the latest version (4.8.6). For correct work of live updates use the latest standard maps available.

Thanks. I can confirm the latest maps for the region dated 31.08.2024 work as intended, even on 4.8.5.

**Comment by JamesBelchamber** [comment]
> It appears that the current solutions are either use Version 1 rendering engine or stop using live updates. Is there another GitHub issue that tracks the live update incompatibility with Version 2 (OpenGL) rendering engine?

Was this issue fixed (and then we had another regression) or had the user's update just become available in the standard map?

I am currently on 5.2.13 and Live Updates only work in Rendering Engine v1. If this is a regression I'm happy to open another ticket :)

**Comment by DmitryAlexei** [comment]
@JamesBelchamber live updates works fine for me using Rendering 2 both in the release and nightly version of the app
please provide coordinates of your changes that are not displayed

**Comment by JamesBelchamber** [comment]
Location: https://osmand.net/map?pin=53.42375,-2.18122#19/53.42375/-2.18122

V1:
<img width="400" height="389" alt="Image" src="https://github.com/user-attachments/assets/a570b369-02f6-4217-ba61-8a76e487e193" />

V2:
<img width="400" height="441" alt="Image" src="https://github.com/user-attachments/assets/5ba0222e-de63-4200-92c0-806fde606b85" />

Turned off backups and wiped storage for the app to ensure it wasn't a weird config thing.

EDIT: Shrunk and cropped images to be less obnoxious..

**Comment by DmitryAlexei** [comment]
Church is displayed after app restart, please try.

https://github.com/user-attachments/assets/03037eb7-9a76-4e39-b41f-67da1cd6f76f

**Comment by JamesBelchamber** [comment]
Sorry, that doesn't help. Tested in Nightly, same behaviour.

**Comment by Rpnpif** [comment]
I have not seen this issue of @JamesBelchamber  on 5.2.13 release. This release is drastically faster and more reliable with OpenGL 2. The CPU or GPU has now a normal temperature. Thank you very much.

**Comment by DmitryAlexei** [comment]
@JamesBelchamber try deleting and redownloading the main map
By the way, I don't see that issue with United Kingdom England North West England map + live updates

**Comment by Nine-Tailed** [comment]
@JamesBelchamber go to the osmand properties in Android and delete the cache.

**Comment by JamesBelchamber** [comment]
None of these helped, but I think I have worked out the issue after testing it on a Pixel phone with the standard OS (I'm running GrapheneOS). This seems to be "caused" by enabling the hardened memory allocator (enabled by default on this OS), and disabling it "fixes" the issue.

OsmAnd should probably investigate this because it implies a (possibly exploitable) memory corruption bug, but at least we have a workaround now.

**Comment by madsamjp** [comment]
On GrapheneOS here. I also have the issue where live updates are not rendered in V2 rendering. Can confirm disabling hardened memory allocator resolves the issue for now. However, this is not ideal from a security perspective.
